src/mfs.jl

```diff
@@ -0,0 +1,239 @@
+export minimal_fatal_shock, MFSBruteForce, MFSBlackBoxOptim
+
+
+"""
+    MFSBruteForce(; kwargs...)
+
+Initialize the optimization algorithm used in `minimal_fatal_shock` function. It is based on randomised search with two consequent steps: random initialization and sphere radius reduction.
+
+## Keyword arguments
+`initial_iterations` number of random pertubations to try in the first step of the algorithm, default = 10000.
+`sphere_iterations` number of steps while initializing random points on hypersphere and decreasing its radius, default = 10000.
+`sphere_decrease_step` factor by which the radius of the hypersphere is decreased, default = 1000.0, the higher the value, 
+the more precise values may be obtained.
+
+"""
+mutable struct MFSBruteForce 
+    initial_iterations::Int64
+    sphere_iterations::Int64
+    sphere_decrease_step::Float64
+end
+
+function MFSBruteForce(; initial_iterations = 10000, sphere_iterations = 10000, sphere_decrease_step = 100.0)
+    MFSBruteForce(initial_iterations, sphere_iterations, sphere_decrease_step)
+end
+
+
+
+"""
+    MFSBlackBoxOptim(; kwargs...)
+
+Initialize the optimization algorithm used in `minimal_fatal_shock` function. It uses BlackBoxOptim.jl package to find the best shock.
+It is based on derivative free optimization and uses the objective function with penalties to find the minimal fatal shock.
+
+## Keyword arguments
+`guess` vector of initial guesses for the optimization algorithm, `default = []`.
+`MaxSteps` maximum number of steps for the optimization algorithm, default = 10000.
+`penalty` penalty value for the objective function, allows to adjust optimization algorithm to find the minimal fatal shock, `default = 1000.0`
+`random_algo` algorithm used to find the initial guess for the optimization algorithm, by default it is initialized as `MFSBruteForce(0,0,0)` and not used. To activate it,
+you need to initialize it with the parameters you want to use, e.g. `MFSBruteForce(1000,1000,100.0)` or `MFSBruteForce()` with default parameters.
+"""
+struct MFSBlackBoxOptim
+    guess::Vector{Float64}
+    MaxSteps::Int64
+    penalty::Float64
+    random_algo::MFSBruteForce
+end
+
+function MFSBlackBoxOptim(; guess = [], MaxSteps = 1000,  penalty = 1000.0, random_algo = MFSBruteForce(0,0,0))
+    MFSBlackBoxOptim(guess, MaxSteps, penalty, random_algo)
+end
+"""
+    minimal_fatal_shock(mapper::AttractorMapper, u0, search_area, algorithm::Union{MFSBruteForce, MFSBlackBoxOptim})  
+
+Runs the minimal fatal shock algorithm on the initial point `u0` and outputs tuple `minimal fatal shock` and its `norm`. Two algorithms are available: [`MFSBruteForce`](@ref) and [`MFSBlackBoxOptim`](@ref).
+[`MFSBruteForce`](@ref) is an algorithm based on randomised search with two consequent steps: random initialization and sphere radius reduction.
+[`MFSBlackBoxOptim`](@ref) is an algorithm based derivative free optimization. It uses BlackBoxOptim package to find the best shock.
```

Comment:
```suggestion
```

You explain the algorithms in their respective docstrings so no reason to repeat here.